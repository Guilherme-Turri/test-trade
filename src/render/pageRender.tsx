import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};