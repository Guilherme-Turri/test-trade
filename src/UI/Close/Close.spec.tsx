import { screen } from '@testing-library/react'
import { Close } from '.'
import { componentRender } from '../../render/componentRender'

describe('</Close>', () => {
  it('should render on screen', () => {
    const { container } = componentRender(<Close />)
    expect(screen.getByTestId('Close')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })
})