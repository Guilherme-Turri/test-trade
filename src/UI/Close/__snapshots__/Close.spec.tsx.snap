// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</Close> should render on screen 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  color: #00c555;
  width: 100%;
  height: 100%;
}

<div>
  <div
    class="c0"
    data-testid="Close"
  >
    <svg
      aria-hidden="true"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        data-name="Layer 2"
      >
        <path
          d="m13.41 12 4.3-4.29a1 1 0 1 0-1.42-1.42L12 10.59l-4.29-4.3a1 1 0 0 0-1.42 1.42l4.3 4.29-4.3 4.29a1 1 0 0 0 0 1.42 1 1 0 0 0 1.42 0l4.29-4.3 4.29 4.3a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42z"
          data-name="close"
        />
      </g>
    </svg>
  </div>
</div>
`;
